android-components/components/browser/state/src/test/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddlewareTest.kt

```diff
@@ -123,79 +142,94 @@ class TranslationsMiddlewareTest {
     }
 
     @Test
-    fun `WHEN OperationRequestedAction is dispatched WITH FETCH_PAGE_SETTINGS AND fetching languages is successful THEN TranslationPageSettings is dispatched`() = runTest {
-        // Mock detected languages are required to pull the language settings
-        val mockFrom = "es"
-        val mockTo = "en"
-        val mockDetectedLanguages = DetectedLanguages(
-            documentLangTag = mockFrom,
-            supportedDocumentLang = true,
-            userPreferredLangTag = mockTo,
-        )
-        val mockState = TranslationsState(
-            translationEngineState = TranslationEngineState(mockDetectedLanguages),
-        )
-        // Had mock issues when mocking anything more granular
-        whenever(store.state.findTab(tab.id)?.translationsState).thenReturn(mockState)
+    fun `WHEN OperationRequestedAction is dispatched WITH FETCH_PAGE_SETTINGS AND fetching settings is successful THEN TranslationPageSettings is dispatched`() = runTest {
+        // Setup
+        setupMockState()
 
-        // Always offer behavior
-        val offerResponse = true
-        whenever(engine.getTranslationsOfferPopup()).thenAnswer { offerResponse }
-
-        // Page language
-        val languageSettingCallback = argumentCaptor<((LanguageSetting) -> Unit)>()
-        val languageResponse = LanguageSetting.ALWAYS
-        whenever(
-            engine.getLanguageSetting(
-                languageCode = any(),
-                onSuccess = languageSettingCallback.capture(),
-                onError = any(),
-            ),
+        val mockPageSettings = TranslationPageSettings(
+            alwaysOfferPopup = true,
+            alwaysTranslateLanguage = true,
+            neverTranslateLanguage = false,
+            neverTranslateSite = true,
         )
-            .thenAnswer {
-                languageSettingCallback.value.invoke(languageResponse)
-            }
 
-        // Never translate site behavior
-        val neverTranslateSiteCallback = argumentCaptor<((Boolean) -> Unit)>()
-        val neverSiteResponse = true
-        whenever(
-            engineSession.getNeverTranslateSiteSetting(
-                onResult = neverTranslateSiteCallback.capture(),
-                onException = any(),
-            ),
-        )
-            .thenAnswer {
-                neverTranslateSiteCallback.value.invoke(neverSiteResponse)
-            }
+        whenever(engine.getTranslationsOfferPopup()).thenAnswer { mockPageSettings.alwaysOfferPopup }
 
-        store.dispatch(
+        // Send Action
+        val action =
             TranslationsAction.OperationRequestedAction(
                 tabId = tab.id,
                 operation = TranslationOperation.FETCH_PAGE_SETTINGS,
-            ),
-        ).joinBlocking()
+            )
 
```

Comment:
nit: maybe remove the new line between declaring the action and invoking it? same for the test below

Reply:
Ty, updated x2!